pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -341,6 +346,11 @@ func ApplyTaskResultsToPipelineResults(
 				continue
 			}
 			variableParts := strings.Split(variable, ".")
+			// if the referenced task is skipped, we should also skip the results replacements
+			// this fixes #6139
```

Comment:
we don't need the reference to the issue in the codebase

Reply:
Thanks! Fixed